torch/_inductor/codegen/cpp_bmm_template.py

```diff
@@ -0,0 +1,171 @@
+# mypy: allow-untyped-defs
+from typing import Callable, List, Optional
+
+from .. import ir
+
+from ..select_algorithm import DataProcessorTemplateWrapper
+from .cpp_gemm_template import CppPackedGemmTemplate, GEMM_TEMPLATE, MICROKERNEL_DEF
+
+from .cpp_template_kernel import CppTemplateKernel
+from .cpp_utils import GemmBlocking
+
+GEMM_SINGLE_THREAD_MM_STUB = r"""
+void single_thread_mm(
+    const {{micro_gemm.get_common_options()['input_t']}}* X,
+    const {{micro_gemm.get_common_options()['input_t']}}* W,
+    {{micro_gemm.get_common_options()['input_t']}}* Y
+    {%- if is_dynamic_M %},
+    const int64_t {{kernel.size(GemmOut, -2, unwrapped=True)}}
```

Comment:
We just need the name of the symbol. The dynamic shapes variables `ks0`, `ks1`, etc. are declared in `kernel.def_kernel()`, so we don't know the name of these symbols beforehand.  So I use the unwrapped calls to just get the name of the symbol.

Reply:
Since it's a formal parameter of function `threaded_mm`, the name doesn't have to be `ks0` or `ks1`, right? We only need to pass in actual parameter of `ks0` when invoking this function? Doesn't it?

Reply:
The problem is I'm trying to fit the `CppBmmTemplate` to work with minimal changes to `CppPackedGemmTemplate`. So in the GEMM template we have `const int64_t M = {{kernel.size(GemmOut, 0)}}`. This evaluates to `const int64_t M = ks0` if only M is dynamic and `const int64_t M = ks1` if M and B are both dynamic. 

So the issue is not that we can't pass this as a parameter, because we could.  I'm trying not to change the code existing in `cpp_gemm_template.py`, so having a version of the size() function that doesn't wrap everything in a static_cast c expression is helpful.

Reply:
I'm a bit confused. The sizes should be automatically generated via `kernel.def_kernel` along with the buffers needed by epilogue fusions. I don't think you have to manually add them.

Reply:
Issue was that `def_kernel` was not being used to create the calls to sub-routines (`single_thread_mm` and `threaded_mm`, so I was constructing those manually. I added a function to CppTemplateKernel `def_kernel_with_name` that lets you create a function definition for any function, not just for the main kernel.